### PR TITLE
Additional Features for Union Pooler Variation Robustness Experiment

### PR DESCRIPTION
**Pull request by rmccall84**
The latest version of the experiment including the variation with a fixed number of substitutions.

### CODE REVIEW

**Comment by rmccall84** [comment]
@scottpurdy please cr

File: union_pooling/union_pooling/experiments/variation_robustness/print_results.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import csv
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/variation_robustness/print_results.py]: add copyright header?

File: union_pooling/union_pooling/experiments/variation_robustness/print_results.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+SIG_FIGS = 5
+
+
+
+def main(dir):
+  fileList = sorted([f for f in listdir(dir) if isfile(join(dir, f))])
+  for f in fileList:
+    with open(join(dir, f), "rU") as outputFile:
+      csvReader = csv.reader(outputFile)
+      next(csvReader)
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/variation_robustness/print_results.py]: `csvReader.next()` is a better IMO

File: union_pooling/union_pooling/experiments/variation_robustness/variation_robustness_experiment.py [hunk 3]
```diff
@@ -54,20 +57,25 @@
...
-changing when sequence actually changes.
+Goal: Characterize the variation robustness of the UnionPooler to various
+perturbations. Explore trade-off between remaining stable to variations yet
+still changing when sequence actually changes.
 """
 
 
 
 _SHOW_PROGRESS_INTERVAL = 3000
+_NO_PERTURBATION = 0
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/variation_robustness/variation_robustness_experiment.py]: You could use an enum class:

```
class PerturbationType(object):
  substitution = 0
  skip = 1
  ...
```

Then check the value as follows:

```
if perturbationType is PerturbationType.substitution:
```

**Comment by scottpurdy** [comment]
Put a few comments, otherwise looks good

**Pull request merged**